src/dbus/server/dbus_thread_object.cpp

```diff
@@ -556,5 +589,248 @@ otError DBusThreadObject::GetSupportedChannelMaskHandler(DBusMessageIter &aIter)
     return err;
 }
 
+otError DBusThreadObject::GetRloc16Handler(DBusMessageIter &aIter)
+{
+    auto     threadHelper = mNcp->GetThreadHelper();
+    otError  err          = OT_ERROR_NONE;
+    uint16_t rloc16       = otThreadGetRloc16(threadHelper->GetInstance());
+
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, rloc16) == OTBR_ERROR_NONE, err = OT_ERROR_INVALID_ARGS);
```

Comment:
`OT_ERROR_INVALID_ARGS` doesn't seem to be the proper error code for this failure. I think we should return `otbrError` for this handlers.

Reply:
The handler function signature has been decided and it will requires quite some effort to change it. I'd prefer to change it after we've unified all the error codes.

Reply:
Another consideration is that in fact all these handlers are calling some OpenThread apis that will return otError. So I decided to let the property handlers return directly an otError to reduce casts.

Reply:
I think we should make a function to cast from `otError` to `otbrError`. In this case, there are errors out of the scope of OT, but related to OTBR (OT error is subset of OTBR error I think). 

- It is okay to leave this to a future PR, but we should create an issue tracking it.
- `OT_ERROR_FAILED` fits better than `OT_ERROR_INVALID_ARGS` for compromised solution :)

Reply:
There is a PR about these error code: https://github.com/openthread/ot-br-posix/pull/373